### DISCUSSION TITLE
Unable to run a forked plugin

### DISCUSSION THREAD

**Discussion by panzacoder** [gatsbyjs/gatsby#35097] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

This is essentially the same issue described in issue #6675.

I have a repo I would like to fork, test in my Gatsby project, and then open a PR back in to the main repo.
Here is my fork: [My Forked Intercom Plugin](https://github.com/panzacoder/gatsby-plugin-intercom-spa).

I can install this version by doing `yarn add panzacoder/gatsby-plugin-intercom-spa` and I can verify that my code is in my repo at `node_modules/gatsby-plugin-intercom-spa` but this code is not executed at all.

I have:
- Confirmed that the original plugin still works
- Attempted sticking the code in /plugins instead as a local plugin (no dice, I think I would need to work on the `package.json` which I have no interest in doing as it's out of the scope of what I'm trying to do
- Put the code directly in my `gatsby-browser.js` and `gatsby-ssr.js` (this works, so I'm 99% confident that my code changes aren't the issue)

### Reproduction Link

https://github.com/panzacoder/gatsby-plugin-intercom-spa

### Steps to Reproduce

1. Have a Gatsby repo
2. `yarn add panzacoder/gatsby-plugin-intercom-spa`
3. Configure plugin according to documentation
4. Nothing happens (no console log, no intercom ping)

### Expected Result

I expect the code in my node_modules to be run when the plugin is configured.

### Actual Result

Nothing happens.

### Environment

```shell
System:
    OS: macOS 12.2.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nodenv/versions/16.13.1/bin/node
    Yarn: 1.22.17 - ~/.nodenv/versions/16.13.1/bin/yarn
    npm: 8.1.2 - ~/.nodenv/versions/16.13.1/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 99.0.4844.51
    Firefox: 85.0
    Safari: 15.3
```

### Config Flags

PRESERVE_FILE_DOWNLOAD_CACHE: true
PARALLEL_SOURCING: true
DEV_SSR: true

**Comment by LekoArts** [comment]
You'll want to use something like https://docs.npmjs.com/cli/v8/commands/npm-link to test out your changes
